code/game/objects/machinery/mortar.dm

```diff
@@ -181,69 +179,27 @@
 			to_chat(user, span_warning("Someone else is currently using [src]."))
 			return
 
-		if(!is_ground_level(z))
-			to_chat(user, span_warning("You cannot fire [src] here."))
-			return
-
-		if(coords["targ_x"] == 0 && coords["targ_y"] == 0) //Mortar wasn't set
-			to_chat(user, span_warning("[src] needs to be aimed first."))
-			return
-
-		var/turf/selfown = locate((coords["targ_x"] + coords["dial_x"]), (coords["targ_y"] + coords["dial_y"]), z)
-		if(get_dist(loc, selfown) < 7)
-			to_chat(user, span_warning("You cannot target this coordinate, it is too close to your mortar."))
-			return
-
-		var/turf/T = locate(coords["targ_x"] + coords["dial_x"] + offset_x, coords["targ_y"]  + coords["dial_x"] + offset_y, z)
-		if(!isturf(T))
-			to_chat(user, span_warning("You cannot fire [src] to this target."))
-			return
-
-		var/area/A = get_area(T)
-		if(istype(A) && A.ceiling >= CEILING_UNDERGROUND)
-			to_chat(user, span_warning("You cannot hit the target. It is probably underground."))
-			return
-
 		user.visible_message(span_notice("[user] starts loading \a [mortar_shell.name] into [src]."),
 		span_notice("You start loading \a [mortar_shell.name] into [src]."))
 		playsound(loc, reload_sound, 50, 1)
 		busy = TRUE
-		if(!do_after(user, 15, TRUE, src, BUSY_ICON_HOSTILE))
+		if(!do_after(user, reload_time, TRUE, src, BUSY_ICON_HOSTILE))
 			busy = FALSE
 			return
 
 		busy = FALSE
 
 		user.visible_message(span_notice("[user] loads \a [mortar_shell.name] into [src]."),
 		span_notice("You load \a [mortar_shell.name] into [src]."))
-		visible_message("[icon2html(src, viewers(src))] [span_danger("The [name] fires!")]")
-		user.transferItemToLoc(mortar_shell, src)
-		playsound(loc, fire_sound, 50, 1)
-		firing = TRUE
-		flick(icon_state + "_fire", src)
-		mortar_shell.forceMove(src)
-
-		if(tally_type == TALLY_MORTAR)
-			GLOB.round_statistics.howitzer_shells_fired++
-			SSblackbox.record_feedback("tally", "round_statistics", 1, "howitzer_shells_fired")		
-		else if(tally_type == TALLY_HOWITZER)
-			GLOB.round_statistics.mortar_shells_fired++
-			SSblackbox.record_feedback("tally", "round_statistics", 1, "mortar_shells_fired")		
-
-		var/turf/G = get_turf(src)
-		G.ceiling_debris_check(2)
-
-		for(var/mob/M in range(7))
-			shake_camera(M, 3, 1)
-		log_game("[key_name(user)] has fired the [src] at [AREACOORD(T)], impact in [travel_time+45] ticks")
-		addtimer(CALLBACK(GLOBAL_PROC, .proc/playsound, T, fall_sound, 50, 1), travel_time)
-		addtimer(CALLBACK(src, .proc/detonate_shell, T, mortar_shell), travel_time + 45)//This should always be 45 ticks!
+		chamber_items.Add(mortar_shell)
+		user.balloon_alert(user, "Right click to fire shell.")
+		qdel(mortar_shell)
```

Comment:
why do you qdel the mortar shell if you JUST put it in the list, this will hard del and add nulls

Reply:
forcemove it and instead qdel after it has fired